monai/networks/nets/mednext.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Portions of this code are derived from the original repository at:
+# https://github.com/MIC-DKFZ/MedNeXt
+# and are used under the terms of the Apache License, Version 2.0.
+
+from __future__ import annotations
+
+import torch
+import torch.nn as nn
+
+from monai.networks.blocks import MedNeXtBlock, MedNeXtDownBlock, MedNeXtUpBlock, OutBlock
+
+
+class MedNeXt(nn.Module):
+    """
+    MedNeXt model class from paper: https://arxiv.org/pdf/2303.09975
+
+    Args:
+        spatial_dims: spatial dimension of the input data. Defaults to 3.
+        init_filters: number of output channels for initial convolution layer. Defaults to 32.
+        in_channels: number of input channels for the network. Defaults to 1.
+        out_channels: number of output channels for the network. Defaults to 2.
+        enc_exp_r: expansion ratio for encoder blocks. Defaults to 2.
+        dec_exp_r: expansion ratio for decoder blocks. Defaults to 2.
+        bottlenec_exp_r: expansion ratio for bottleneck blocks. Defaults to 2.
+        kernel_size: kernel size for convolutions. Defaults to 7.
+        deep_supervision: whether to use deep supervision. Defaults to False.
+        do_res: whether to use residual connections. Defaults to False.
+        do_res_up_down: whether to use residual connections in up and down blocks. Defaults to False.
+        blocks_down: number of blocks in each encoder stage. Defaults to [2, 2, 2, 2].
+        blocks_bottleneck: number of blocks in bottleneck stage. Defaults to 2.
+        blocks_up: number of blocks in each decoder stage. Defaults to [2, 2, 2, 2].
+        norm_type: type of normalization layer. Defaults to 'group'.
+        grn: whether to use Global Response Normalization (GRN). Defaults to False.
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int = 3,
+        init_filters: int = 32,
+        in_channels: int = 1,
+        out_channels: int = 2,
+        enc_exp_r: int = 2,
+        dec_exp_r: int = 2,
+        bottlenec_exp_r: int = 2,
```

Comment:
IMO encoder_expansion_ratio (the same for decoder and bottleneck) is better here. It's a public API, makes it easier to see what it refers to without having to look at the docs and the few letters more don't matter that much